src/test/java/com/wikia/webdriver/elements/mercury/pages/CategoryPage.java

```diff
@@ -0,0 +1,188 @@
+    String currentUrl = driver.getCurrentUrl();
+    Assertion.assertTrue(
+        currentUrl.contains(expectedUrl),
+        String.format("Expected part \"%s\" was not found in \"%s\".", expectedUrl, currentUrl)
+    );
+    PageObjectLogging.logInfo(String.format("You were redirected to page: \"%s\".", name));
+
+    return this;
+  }
+
+  private CategoryPage articleContainerIsVisible() {
+    wait.forElementVisible(article);
+    PageObjectLogging.logInfo("Article container is visible.");
+
+    return this;
+  }
+
+  private CategoryPage articleContainerIsNotPresent() {
+    wait.forElementNotPresent(article);
+    PageObjectLogging.logInfo("Article container is not present in DOM, which is nice.");
+
+    return this;
+  }
+
+  private CategoryPage categorySectionsContainerIsVisible() {
+    wait.forElementVisible(categorySections);
+    PageObjectLogging.logInfo("Category sections container is visible.");
+
+    return this;
+  }
+
+  private CategoryPage noMembersMessageIsVisible() {
+    wait.forElementVisible(noMembersMessage);
+    PageObjectLogging.logInfo("Info message about no pages in category is visible.");
+
+    return this;
+  }
+
+  private CategoryPage navigateThroughSection(String name, By buttonSelector) {
+    WebElement section = driver.findElement(
+        By.cssSelector(String.format("#%s.category-section", name))
+    );
+    wait.forElementVisible(section);
+    PageObjectLogging.logInfo(String.format("Category section \"%s\" is visible.", name));
+
+    WebElement sectionBatchFirstItem = section.findElement(By.cssSelector("li"));
+    wait.forElementVisible(sectionBatchFirstItem);
+    String firstItemTextInFirstBatch = sectionBatchFirstItem.getText();
+
+    WebElement buttonElement = section.findElement(buttonSelector);
+    wait.forElementClickable(buttonElement);
+    buttonElement.click();
+    waitForBatchToBeLoaded(section);
+
+    sectionBatchFirstItem = section.findElement(By.cssSelector("li"));
+    wait.forElementVisible(sectionBatchFirstItem);
+    String firstItemTextInSecondBatch = sectionBatchFirstItem.getText();
+
+    Assertion.assertNotEquals(
+        firstItemTextInFirstBatch,
+        firstItemTextInSecondBatch,
+        "First element of the newly loaded section is the same as the previous one."
+    );
+    PageObjectLogging.logInfo("New section was loaded correctly.");
+
+    return this;
+  }
+
+  private void waitForBatchToBeLoaded(WebElement section) {
+    int attemptLimit = 5;
+    boolean isLoadingBatchPresent = true;
+
+    PageObjectLogging.logInfo("Waiting for loading-batch to be present");
+
+    try {
+      new WebDriverWait(driver, attemptLimit, 200).until(new ExpectedCondition<Boolean>() {
+        @Nullable
+        @Override
+        public Boolean apply(@Nullable WebDriver input) {
+          return section.getAttribute("class").contains("loading-batch");
+        }
+      });
+    } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)